address-book-client/src/js/views/AddEditAddressModal.js

```diff
@@ -0,0 +1,140 @@
+    this.relationError = this.relationInput.nextElementSibling;
+    this.phoneInput = this.modalEl.phone;
+    this.phoneError = this.phoneInput.nextElementSibling;
+    this.emailInput = this.modalEl.email;
+    this.emailError = this.emailInput.nextElementSibling;
+    this.avatarInput = this.modalEl.avatar;
+    this.avatarError = this.avatarInput.nextElementSibling;
+
+    this.modalEl.addEventListener("submit", this.handleSubmit.bind(this));
+  }
+
+  // Handle open ADD-modal
+  handleOpenAddModal() {
+    this.modalEl.classList.add("modal--active");
+    this.overlayEl.classList.add("overlay--active");
+    this.openEvent.trigger();
+  }
+
+  // Handle close ADD-modal
+  handleCloseAddModal() {
+    this.modalEl.classList.remove("modal--active");
+    this.overlayEl.classList.remove("overlay--active");
+
+    // Clear any previous error messages and styles
+    this.resetModal();
+    this.modalEl.reset();
+
+    this.closeEvent.trigger();
+  }
+
+  handleSubmit(e) {
+    e.preventDefault();
+    this.validateForm();
+    this.formEvent.trigger();
+  }
+
+  //----- VALIDATE FORM -----//
+
+  validateForm() {
+    // Get values from the input fields
+    const modalEl = document.querySelector(".modal");
+    const nameInput = modalEl.name;
+    const nameError = nameInput.nextElementSibling;
+    const relationInput = modalEl.relation;
+    const relationError = relationInput.nextElementSibling;
+    const phoneInput = modalEl.phone;
+    const phoneError = phoneInput.nextElementSibling;
+    const emailInput = modalEl.email;
+    const emailError = emailInput.nextElementSibling;
+    const avatarInput = modalEl.avatar;
+    const avatarError = avatarInput.nextElementSibling;
+
+    let isValid = true;
+
+    this.resetModal();
+
+    // Object to store field validation data
+    const fields = [
+      { name: "name", regex: REGEX.NAME, error: nameError, requiredMessage: MESSAGE.NAME_REQUIRED, invalidMessage: MESSAGE.INVALID_NAME },
+      { name: "relation", regex: REGEX.NAME, error: relationError, requiredMessage: MESSAGE.RELATION_REQUIRED, invalidMessage: MESSAGE.INVALID_RELATION },
+      { name: "phone", regex: REGEX.PHONE, error: phoneError, requiredMessage: MESSAGE.PHONE_REQUIRED, invalidMessage: MESSAGE.INVALID_PHONE },
+      { name: "email", regex: REGEX.EMAIL, error: emailError, requiredMessage: MESSAGE.EMAIL_REQUIRED, invalidMessage: MESSAGE.INVALID_EMAIL },
+      { name: "avatar", regex: REGEX.AVATAR, error: avatarError, requiredMessage: MESSAGE.AVATAR_REQUIRED, invalidMessage: MESSAGE.INVALID_AVATAR },
+    ];
+
+    // Loop through each field to perform validation
+    for (const field of fields) {
+      const inputEl = modalEl[field.name];
+      const errorEl = field.error;
+      const value = inputEl.value;
+
+      const isValidField = field.regex.test(value);
+
+      // Check if the field value is empty
+      if (value.trim() === "") {
+        inputEl.classList.add("input--warning");
```

Comment:
Can create a method for this. i.e. this.showErrorMessage(inputElement, errorElement, errorMessage)
inputEl.classList.add("input--warning");
 errorEl.textContent = field.requiredMessage;
 errorEl.classList.add("warning-text--active");